src/core/AutoRest.Core/Settings.cs

```diff
@@ -81,6 +82,13 @@ public Settings()
         public string Input { get; set; }
 
         /// <summary>
+        /// Gets or sets a name for the generated client models Namespace and Models output folder
+        /// </summary>
+        [SettingsInfo("Name to use for the generated client models namespace and folder name. Not supported by all code generators.")]
```

Comment:
I'm wondering if we _want_ to force the `models` folder as a sub-folder/sub-namespace of the parent (ie, by doing it as you've done here, declared as just the last `.part` ), or if we should allow them to specify the whole namespace/foldername. (ie, `-ModelsName Foo.Bar.Bin.Baz` ) 

That would require the code generator to calculate the folder name a bit more carefully  (ie, generated relative between the `client` namespace/path and the `models` namespace/path.

And does this let us choose the same namespace for the client as the model itself? Is it possible that someone **would** want that?

(food for thought...)